src/data-model-providers/codegen/CodegenDataModelProvider.cpp

```diff
@@ -43,203 +47,221 @@
+    if (interface != nullptr)
     {
-#if CHIP_CONFIG_DATA_MODEL_EXTRA_LOGGING
-        ChipLogError(AppServer, "Failed to get data version for %d/" ChipLogFormatMEI, static_cast<int>(path.mEndpointId),
-                     ChipLogValueMEI(cluster.clusterId));
-#endif
-        return CHIP_ERROR_NOT_FOUND;
-    }
+        size_t commandCount = 0;
 
-    DataModel::ClusterInfo info(*versionPtr);
-    // TODO: set entry flags:
-    //   info->flags.Set(ClusterQualityFlags::kDiagnosticsData)
-    return info;
-}
+        CHIP_ERROR err = interface->EnumerateGeneratedCommands(
+            path,
+            [](CommandId id, void * context) -> Loop {
+                *reinterpret_cast<size_t *>(context) += 1;
+                return Loop::Continue;
+            },
+            reinterpret_cast<void *>(&commandCount));
 
-/// Converts a EmberAfCluster into a ClusterEntry
-std::variant<CHIP_ERROR, DataModel::ClusterEntry> ClusterEntryFrom(EndpointId endpointId, const EmberAfCluster & cluster)
-{
-    ConcreteClusterPath clusterPath(endpointId, cluster.clusterId);
-    auto info = LoadClusterInfo(clusterPath, cluster);
+        if (err == CHIP_NO_ERROR)
+        {
+            ReturnErrorOnFailure(result.reserve(commandCount));
 
-    if (CHIP_ERROR * err = std::get_if<CHIP_ERROR>(&info))
-    {
-        return *err;
+            using EnumerationData = struct
+            {
+                DataModel::MetadataList<CommandId> generatedCommandList;
+                CHIP_ERROR processingError;
+            };
+            EnumerationData enumerationData;
+            enumerationData.processingError = CHIP_NO_ERROR;
+
+            ReturnErrorOnFailure(interface->EnumerateGeneratedCommands(
+                path,
+                [](CommandId id, void * context) -> Loop {
+                    auto input = reinterpret_cast<EnumerationData *>(context);
+
+                    CHIP_ERROR appendError = input->generatedCommandList.Append(id);
+                    if (appendError != CHIP_NO_ERROR)
+                    {
+                        input->processingError = appendError;
+                        return Loop::Break;
+                    }
+                    return Loop::Continue;
+                },
+                reinterpret_cast<void *>(&result)));
+            ReturnErrorOnFailure(enumerationData.processingError);
+
+            // the two invocations MUST return the same sizes.
+            VerifyOrReturnError(enumerationData.generatedCommandList.size() == commandCount, CHIP_ERROR_INTERNAL);
+
+            result = std::move(enumerationData.generatedCommandList);
+            return CHIP_NO_ERROR;
+        }
+        VerifyOrReturnError(err == CHIP_ERROR_NOT_IMPLEMENTED, err);
     }
 
-    if (DataModel::ClusterInfo * infoValue = std::get_if<DataModel::ClusterInfo>(&info))
+    if ((serverCluster == nullptr) || (serverCluster->generatedCommandList == nullptr))
+    {
+        return {};
```

Comment:
That's returning `{}` as CHIP_ERROR?

Reply:
copy and paste from when we return metadata list. Fixed.